module/category-tree/tests/Domain/Entity/CategoryTreeTest.php

```diff
@@ -67,4 +68,43 @@ public function testAddingCategory(): void
         $tree->addCategory($category, $root);
         $this->assertTrue($tree->hasCategory($category));
     }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     */
+    public function testAddingExistingCategory(): void
+    {
+        $category = CategoryId::generate();
+
+        $tree = new CategoryTree($this->id, $this->code, $this->name);
+
+        $tree->addCategory($category);
+        $tree->addCategory($category);
+    }
+
+    /**
+     */
+    public function testRenamingCategory(): void
+    {
+        $tree = new CategoryTree($this->id, $this->code, $this->name);
+
+        $name2 = new TranslatableString(['en' => 'english']);
+        $tree->changeName($name2);
+        $this->assertEquals($name2, $tree->getName());
+    }
+
+    /**
+     */
+    public function testUpdatingCategories(): void
+    {
+        $categoryId = CategoryId::generate();
+        $node = new Node($categoryId);
+        $categories = [$node];
+
+        $tree = new CategoryTree($this->id, $this->code, $this->name);
+
+        $tree->updateCategories($categories);
+
+        $this->assertTrue($tree->hasCategory($categoryId));
```

Comment:
Add test that check is categorises correct after update, this one only check are any categories, but not correct corrct

Reply:
I am adding particular cattegory with Category Id and then checking if this particular category exist in CategoryTree by puting CategoryId (->hasCategory(CategoryId)